netcore/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -236,10 +237,23 @@ internal static ulong GetSegmentSize ()
 			return 1024 * 1024 * 16;
 		}
 
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		static extern GCMemoryInfo _GetGCMemoryInfo (out long highMemoryLoadThresholdBytes,
+														     out long memoryLoadBytes,
+															 out long totalAvailableMemoryBytes,
+															 out long heapSizeBytes,
+															 out long fragmentedBytes); 
+
 		public static GCMemoryInfo GetGCMemoryInfo ()
 		{
-			// TODO: Implement
-			return default;
+			_GetGCMemoryInfo(out long highMemoryLoadThresholdBytes,
+							 out long memoryLoadBytes,
+							 out long totalAvailableMemoryBytes,
+							 out long heapSizeBytes,
+							 out long fragmentedBytes );
+			
```

Comment:
I have seen native code repeatedly fail to initialize `out` parameters.
Yet safety depends on it perhaps.
So personally I always use `ref` and never `out`.
Opinions may vary.
This is not going to be fast in any event.